Bookish.Web/Controllers/HomeController.cs

```diff
@@ -65,12 +69,24 @@ public IActionResult NewBookPost(string title, string author, string isbn, int t
             }
 
             bookishService.AddBook(title, author, isbn, totalCopies);
-            return RedirectToAction("Barcodes");
+            return RedirectToAction("Barcodes", new { isbn });
         }
 
-        public IActionResult Barcodes()
+        [Route("/Home/Barcodes/{isbn}")]
+        public IActionResult Barcodes(string isbn)
         {
-            return View();
+            var bookCopies = bookishService.GetCopies(isbn).ToList();
```

Comment:
We want to have as little logic in the controller as possible. Therefore instead of using two services, the barcode service should be using the bookish service. So here we should just need to call one method on barcode service to get all the data we need 🙂

Reply:
I'm not sure I understand, can you elaborate please?

Reply:
So I would expect you to call one method here. Something like `barcodeService.getNewBooksWithBarcodes(isbn)` or `barcodeService.getBarcodes(isbn)` and that should return everything the view model will need. In this case I think that is a book title, author and a list of encoded barcode strings. 

So I'd expect your barcode service method to:
1. Call a method on the bookish service to get the book copies.
2. Create a list of encoded barcode strings.
3. Return that information in some form, e.g. a NewBook object which has a title, author and list of encoded barcode strings. Or you could return a list of NewBook objects where each object has a title author and encoded barcode string. Either is fine 🙂 

Then the logic here in the controller will be minimal, just getting the data from the service and passing it to the view model, which is what I'd expect. Oh and it should check that there actually are any copies of the book (by just cheching that the result from the barcode service isn't empty) and redirect to a 404 if not. Does that make sense? 🙂